daemon/create.go

```diff
@@ -150,6 +150,14 @@ func (daemon *Daemon) generateSecurityOpt(ipcMode containertypes.IpcMode, pidMod
 
 		return label.DupSecOpt(c.ProcessLabel), nil
 	}
+	if pidContainer := pidMode.Container(); pidContainer != "" {
+		c, err := daemon.GetContainer(pidContainer)
+		if err != nil {
+			return nil, err
+		}
+
+		return label.DupSecOpt(c.ProcessLabel), nil
```

Comment:
In order for two containers to share the same PID or same IPC Namespace, they have to run with the same SELinux label.  Otherwise SELinux will break when a process in container1 tries to look at the processes in container2, or a process in container 1 tries to use an IPC created in container2.  This is why when the second container is created it switches to the first containers labels.  By merging the containers namespaces, you are saying that they share the same security structure, from an SELinux point of view.